feat(admin-ui): Filter reservations

pKorsholm opened the pull request:
**What**
- Introduce reservation filters in admin

Fixes CORE-1387

adrien2p commented on packages/medusa/src/api/routes/admin/reservations/create-reservation.ts:
```diff
@@ -121,6 +121,7 @@ export default async (req, res) => {
  */
 export class AdminPostReservationsReq {
   @IsString()
+  @IsOptional()
```
question: Does that mean that we support reservations that are not related to an item?

pKorsholm replied:
Yes, "Custom reservations" that can exist outside of orders

adrien2p commented on packages/medusa/src/api/routes/admin/reservations/index.ts:
```diff
@@ -67,6 +69,25 @@ export type AdminReservationsRes = {
+ *         description: inventory item from inventory module
+ *         $ref: "#/components/schemas/InventoryItemDTO"
+ */
+export type ExtendedReservationItem = ReservationItemDTO & {
```
suggestion: Should we move that to the types package?

adrien2p commented on packages/medusa/src/api/routes/admin/reservations/list-reservations.ts:
```diff
@@ -110,12 +116,60 @@ import {
+    listConfig.relations = listConfig.relations?.filter((r) =>
+      relationsPredicates.every((p) => p(r))
+    )
+  }
```
todo: From my understanding, we just want to remove line_item and inventory_item from the relations, in that case I would suggest something simpler which avoid looping through the predicates and looping through the relations as well as making it more readable and easy to update if needed

```ts
const relations = new Set(listConfig.relations ?? [])

const includeItems = relations.delete("line_item")
const includeInventoryItems = relations.delete("inventory_item")

if (listConfig.relations?.length) {
  listConfig.relations = [...relations]
}

// ...
```

Wdyt?

adrien2p replied:
One stone two birds 😅 let me know what you think

pKorsholm replied:
sure

adrien2p commented on packages/medusa/src/api/routes/admin/reservations/utils/join-inventory-items.ts:
```diff
@@ -0,0 +1,35 @@
+
+  const inventoryItemMap = new Map(inventoryItems.map((i) => [i.id, i]))
+
+  return await Promise.all(
```
todo: rm Promise.all ans just a non async map should suffice

adrien2p commented on packages/medusa/src/api/routes/admin/reservations/utils/join-line-items.ts:
```diff
@@ -0,0 +1,33 @@
+  const lineItemMap = new Map(lineItems.map((i) => [i.id, i]))
+
+  return await Promise.all(
+    reservations.map(async (reservation) => {
```
todo: same as the previous comment

adrien2p reviewed (changes requested):
I ve only reviewed the backenmd for now

pKorsholm commented:
@adrien2p I addressed your feedback in #4081 as that's got to do with the backend and this should only be the admin ui. I think updating the other branch pushed a bunch of unrelated changes to the files, but your comments are fixed in the other pr.

adrien2p commented:
> @adrien2p I addressed your feedback in #4081 as that's got to do with the backend and this should only be the admin ui. I think updating the other branch pushed a bunch of unrelated changes to the files, but your comments are fixed in the other pr.

sounds good 🚀

olivermrbl commented on packages/admin-ui/ui/src/components/atoms/date-picker/date-picker.tsx:
```diff
@@ -11,10 +11,11 @@ import InputHeader from "../../fundamentals/input-header"
 import { DateTimePickerProps } from "./types"
 
-const getDateClassname = (d: Date, tempDate: Date) => {
+const getDateClassname = (d: Date, tempDate: Date, greyPast: boolean) => {
```
todo: More intuitive naming for `greyPast`. Think you will need the context to understand this one.

olivermrbl replied:
Should this have a default value assigned here instead of in the calendar component to stay backward compatible?

pKorsholm replied:
went for `greyPastDates`, wdyt? the notion is a bit hard to communicate but I see the confusion that could arise with just `greyPast`.

Assigned default values here and in the component 😄

olivermrbl commented on packages/admin-ui/ui/src/components/templates/reservations-table/index.tsx:
```diff
@@ -222,16 +289,19 @@ const ReservationsTable: React.FC<ReservationsTableProps> = () => {
       >
         <Table
-          enableSearch
+          // enableSearch
```
todo:
```suggestion
```

olivermrbl commented on packages/admin-ui/ui/src/components/molecules/select/next-select/components/single-value.tsx:
```diff
@@ -13,28 +16,50 @@ const SingleValue = <
+    return (e.offsetParent as HTMLDivElement).offsetWidth < e.scrollWidth
+  }
+
+  const toolTip = !getValue().length
+    ? null
+    : ((getValue()?.[0] as { label: string })?.label as string) ?? null
```
comment: This looks a bit funky. If possible, we should make this a tiny bit more readable 😄

pKorsholm replied:
created a short util method instead to extract this in the component for readability

olivermrbl reviewed (changes requested):
epic work @pKorsholm !! 🚀 

only minuscule todos

olivermrbl reviewed (changes requested):
epic work @pKorsholm !! 🚀 

only minuscule todos

olivermrbl commented:
Also, missing a changeset :)

olivermrbl commented:
@adrien2p will you check if your requested changes have been addressed?
